app/code/Magento/CatalogGraphQl/Model/Resolver/Category/SortFieldsOptions.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogGraphQl\Model\Resolver\Category;
+
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+
+/**
+ * Retrieves the sort fields options information object
+ */
+class SortFieldsOptions implements ResolverInterface
+{
+    /**
+     * @var ValueFactory
+     */
+    private $valueFactory;
+
+    /**
+     * @var \Magento\Catalog\Model\Config
+     */
+    private $catalogConfig;
+
+    /**
+     * @param ValueFactory $valueFactory
+     * @param \Magento\Catalog\Model\Config $catalogConfig
+     */
+    public function __construct(
+        ValueFactory $valueFactory,
+        \Magento\Catalog\Model\Config $catalogConfig
+    ) {
+        $this->valueFactory = $valueFactory;
+        $this->catalogConfig = $catalogConfig;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null) : Value
+    {
+        $sortFieldsOptions = [
```

Comment:
Please use `\Magento\Catalog\Model\Category\Attribute\Source\Sortby::getAllOptions`. It has two benefits:
- includes position, so no need to hardcode it here again
- result is already properly formatted, no need to do foreach (assuming that `key` is renamed to `value` as suggested in another comment)

Reply:
How about this one `\Magento\Catalog\Model\Config::getAttributeUsedForSortByArray`?

Reply:
It's the same, this function also contains translation function for Position option
``` $options = ['position' => __('Position')];```

Reply:
The problem is that underlying GraphQL library (webonyx) expects exact match between declared type (string) and the actual value type (Magento\Framework\Phrase), and it does not try to perform type casting. 

I was able to make it work by manually casting Phrase to string before returning the result:
```
        $sortFieldsOptions = $this->sortBy->getAllOptions();
        array_walk(
            $sortFieldsOptions,
            function (&$option) {
                $option['label'] = (string)$option['label'];
            }
        );
```